RPC.en.rst

```diff
@@ -3,59 +3,257 @@
 .. highlight:: cpp
 .. default-domain:: cpp
 
-.. _rpc_refactor:
+.. _rpc:
 
-RPC Refactor
+RPC
 *************
 
+Overview
+========
+
 To communicate between ``traffic_manager`` and ``traffic_server`` there is an RPC mechanism in
-``mgmt/api``. This is a simple serialization style RPC which runs over sockets. Current the logic is
-tightly embedded in the management library. It would be useful for various reasons to extract this
-in to a separate library.
+``mgmt/``. This is a simple serialization style RPC which runs over sockets.
+
+
+Serialization Mechanism
+=======================
+
+A RPC message is sent as a ``MgmtMessageHdr`` followed by the serialized data in bytes. 
+
+.. class:: MgmtMarshall
+
+   This is the class used to marshall data objects. It provides functions to marshall and unmarshall data. Each data object is associated with a field. Fields can be:
+
+.. enum:: MGMT_MARSHALL_INT
```

Comment:
Unfortunately enum support is rather lacking. I generally use `c:macro::` instead for the values and `type::` for the enum itself. Otherwise use `:cpp:enum` for the type and indented `:cpp:enumerator` for the values.

Reply:
changed.